ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -87,15 +105,15 @@ class StoreKit1IntegrationTests: BaseIntegrationTests {
         await self.assertSnapshot(data)
     }
 
+    @available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, visionOS 1.0, *)
     func testPurchaseFailuresAreReportedCorrectly() async throws {
-        self.testSession.failTransactionsEnabled = true
-        self.testSession.failureError = .invalidSignature
+        try await self.testSession.setSimulatedError(.purchase(.purchaseNotAllowed), forAPI: .purchase)
```

Comment:
This test has been failing on iOS 17 for a while now since the `failureError` API has been deprecated. After talking with @MarkVillacampa about this offline, we decided to use the new `setSimulatedError()` API here and test for a `purchaseNotAllowedError` instead, since the `setSimulatedError` returns an unknown `StoreKitError` when making a purchase.